### PR TITLE
[processor/deltatocumulativeprocessor]: fix incorrect bucket counts when downscaling exp histograms

### PR DESCRIPTION
**Pull request by edma2**
**Description:** <Describe what has changed.>

The downscaling algorithm collapses bucket by pairing neighbors into new buckets. The simple case occurs when there is an even number of buckets and offset is zero or even.
```
1 1 1 1
 2   2
```

When there is an odd number of buckets, the last bucket is accounted for by incrementing the `size` variable to ensure an extra loop iteration.
```
1 1 1
 2   1
```

When offset is odd, the first bucket is accounted for by special handling in the loop.
```
ø 1 1 1
 1   2
```

However, when there is an even number of buckets, AND an odd offset, this happens:
```
ø 1 1
 1   0
```

The collapsed buckets are incorrect because the buckets sum up to 1 (1+0) instead of 2 (1+1).

Handle this edge case by incrementing the `size` variable when offset is odd
```
ø 1 1
 1   1
```

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

Added some test cases to showcase the bug.

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by RichieSams — APPROVED** [review]

**Review by jesusvazquez — APPROVED** [review]
LGTM

**Review by tombrk — APPROVED** [review]
@edma2 please add a changelog entry: https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#adding-a-changelog-entry

otherwise LGTM

File: processor/deltatocumulativeprocessor/internal/data/expo/scale_test.go [hunk 1]
```diff
@@ -45,6 +45,18 @@ func TestDownscale(t *testing.T) {
...
+		{scale: 1, bkt: "ø 1 1 0"},
+		{scale: 0, bkt: " 1   1"},
+	}, {
+		{scale: 1, bkt: "ø 1 1"},
+		{scale: 0, bkt: " 1   1"},
+	}, {
+		{scale: 1, bkt: "-  1 1"},
+		{scale: 0, bkt: "- 1   1"},
+	}, {
+		{scale: 5, bkt: "- 5 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 9 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 3 0 0 0 0 0 0 0 0 0 0 0 0 5 0 0 0 0 0 0 0 0 0 0 5 0 0 0 0 0 0 0 3 0 0 0 0 0 0 8 0 0 0 0 0 6 0 0 0 0 0 6 0 0 0 0 5 0 0 0 5 0 0 0 2 0 0 0 5 0 0 7 0 0 0 4 0 5 0 0 4 0 0 3 0 3 0 0 8"},
```
Comment (jpkrohling) [inline processor/deltatocumulativeprocessor/internal/data/expo/scale_test.go:58]: The previous ones make sense to me, but I have to admit that I have no idea about this and the next test case. I'm trusting the folks reviewing this before me and approving this PR.

**Review by jpkrohling — APPROVED** [review]

**Comment by edma2** [comment]
@sh0rez thanks, added a changelog entry

File: processor/deltatocumulativeprocessor/internal/data/expo/scale_test.go (hunk 1 above)
Comment (sh0rez) [reply processor/deltatocumulativeprocessor/internal/data/expo/scale_test.go:58]: agreed, this while I _think_ its correct this case is overly complex and as such of limited use. can we come up with a simpler one instead?

Comment (edma2) [reply processor/deltatocumulativeprocessor/internal/data/expo/scale_test.go]: done

**Review by jpkrohling — APPROVED** [review]

**Comment by edma2** [comment]
@sh0rez @jpkrohling thanks for your reviews and approvals, can we merge it now?

**Review by tombrk — APPROVED** [review]

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**